lib/polymorphic_integer_type/extensions.rb

```diff
@@ -58,6 +58,8 @@ def remove_type_and_establish_mapping(name, options, scope)
 
           options[:foreign_key] ||= "#{poly_type}_id"
           foreign_type = options.delete(:foreign_type) || "#{poly_type}_type"
+          options[:foreign_integer_type] = foreign_type
```

Comment:
This seems like a fairly strange way to pass data back.  Like, we're providing a hash, we're modifying the hash, we're then removing the "information" passed back in our call to super, and then we're using that information.

Can we simplify that?

Reply:
I can pass those as extra return?
but I follow this pattern basically because it's happening everywhere https://github.com/clio/polymorphic_integer_type/blob/master/lib/polymorphic_integer_type/extensions.rb#L39
but saying it's the best way though.

Reply:
I think there's a difference between removing options, and adding them in, only to be later removed.

Let's return them separately.